### PR TITLE
docs: update navigation for version 9 related information

### PR DESCRIPTION
**Pull request by aikithoughts**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Migration information is unavailable in the Table of Contents

Issue Number: N/A

## What is the new behavior?

Migration information is now available in the Table of Contents.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
Based on #34979

### CODE REVIEW

File: aio/angular.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
     "packageManager": "yarn",
     "warnings": {
       "typescriptMismatch": false
-    }
+    },
+    "analytics": false
```
Comment (kara) [inline aio/angular.json]: Revert? Doesn't seem related to your changes.

File: aio/content/navigation.json [hunk 2]
```diff
@@ -704,9 +704,56 @@
...
+                  "tooltip": "Migration from the deprecated Renderer API to the newer Renderer2 API."
+                },
+                {
+                  "url": "guide/migration-dynamic-flag",
+                  "title": "Dynamic Queries Flag",
+                  "tooltip": "In @ViewChild() and @ContentChild() version 9, default value kept static: false."
+                },
+                {
+                  "url": "guide/migration-injectable",
+                  "title": "Migration @Injectable() Decorators",
```
Comment (kara) [inline aio/content/navigation.json]:
```suggestion
                  "title": "Missing @Injectable() Decorators",
```

File: aio/content/navigation.json [hunk 3]
```diff
@@ -704,9 +704,56 @@
...
+              "children": [
+                {
+                  "url": "guide/migration-renderer",
+                  "title": "Renderer to Renderer2",
+                  "tooltip": "Migration from the deprecated Renderer API to the newer Renderer2 API."
+                },
+                {
+                  "url": "guide/migration-dynamic-flag",
+                  "title": "Dynamic Queries Flag",
+                  "tooltip": "In @ViewChild() and @ContentChild() version 9, default value kept static: false."
```
Comment (kara) [inline aio/content/navigation.json]:
```suggestion
                  "tooltip": "Migration to remove unnecessary`static: false` flag from @ViewChild and @ContentChild queries."
```

File: aio/content/navigation.json [hunk 4]
```diff
@@ -704,9 +704,56 @@
...
+                },
+                {
+                  "url": "guide/migration-injectable",
+                  "title": "Migration @Injectable() Decorators",
+                  "tooltip": "Migration for missing @Injectable() decorators and incomplete provider definitions"
```
Comment (kara) [inline aio/content/navigation.json]:
```suggestion
                  "tooltip": "Migration to add missing @Injectable() decorators and correct incomplete provider definitions"
```

File: aio/content/navigation.json [hunk 5]
```diff
@@ -704,9 +704,56 @@
...
+                },
+                {
+                  "url": "guide/migration-localize",
+                  "title": "$localize Global Import",
+                  "tooltip": "Adds an import statement for @angular/localize to polyfills.ts"
```
Comment (kara) [inline aio/content/navigation.json]:
```suggestion
                  "tooltip": "Migration to add an import statement for @angular/localize to polyfills.ts"
```

File: aio/content/navigation.json [hunk 6]
```diff
@@ -704,9 +704,56 @@
...
+                },
+                {
+                  "url": "guide/migration-module-with-providers",
+                  "title": "ModuleWithProviders",
+                  "tooltip": "Added a generic type to any ModuleWithProviders usages that are missing the generic"
```
Comment (kara) [inline aio/content/navigation.json]:
```suggestion
                  "tooltip": "Migration to add a generic type to any ModuleWithProviders usages that are missing the generic"
```

File: aio/content/navigation.json [hunk 7]
```diff
@@ -704,9 +704,56 @@
...
+                  "tooltip": "Migration for missing @Injectable() decorators and incomplete provider definitions"
+                },
+                {
+                  "url": "guide/migration-localize",
+                  "title": "$localize Global Import",
+                  "tooltip": "Adds an import statement for @angular/localize to polyfills.ts"
+                },
+                {
+                  "url": "guide/migration-module-with-providers",
+                  "title": "ModuleWithProviders",
```
Comment (kara) [inline aio/content/navigation.json]:
```suggestion
                  "title": "Missing ModuleWithProviders Generic",
```

File: aio/content/navigation.json [hunk 8]
```diff
@@ -704,9 +704,56 @@
...
+                  "tooltip": "Added a generic type to any ModuleWithProviders usages that are missing the generic"
+                },
+                {
+                  "url": "guide/migration-undecorated-classes",
+                  "title": "@Directive()/@Component() decorator migration",
```
Comment (kara) [inline aio/content/navigation.json]:
```suggestion
                  "title": "Missing @Directive()/@Component() Decorators",
```

File: aio/content/navigation.json [hunk 9]
```diff
@@ -704,9 +704,56 @@
...
+                  "tooltip": "@Directive()/@Component() migration to the latest version."
```
Comment (kara) [inline aio/content/navigation.json]:
```suggestion
                  "tooltip": "Migration to add missing @Directive()/@Component() decorators"
```

**Review by kara — CHANGES_REQUESTED** [review]
I like the way the nav looks now with the CSS changes! Some comments about the tooltips below (adding more context, etc)

File: aio/angular.json (hunk 1 above)
Comment (IgorMinar) [reply aio/angular.json]: +1

File: aio/angular.json [hunk 10]
```diff
@@ -192,4 +193,4 @@
     }
   },
   "defaultProject": "site"
-}
+}
```
Comment (IgorMinar) [inline aio/angular.json]: please revert

**Review by IgorMinar — CHANGES_REQUESTED** [review]
the 4th level is really pushing it :-/

Some of these menu items look quite bad when they stretch 2-3 lines. Can we find a way to make these fit on a single line (reduce font / reduce text)?

Screen Shot 2020-02-04 at 8 47 32 PM

File: aio/content/navigation.json [hunk 11]
```diff
@@ -704,9 +704,56 @@
           "tooltip": "Angular versioning, release, support, and deprecation policies and practices."
         },
         {
-          "url": "guide/updating-to-version-9",
           "title": "Updating to Version 9",
-          "tooltip": "Support for updating your application from version 8 to 9."
+          "tooltip": "Support for updating your application from version 8 to 9.",
```
Comment (IgorMinar) [inline aio/content/navigation.json]: @aikidave somehow your editor is configured in a way that causes the files you touch to be reformatted. I don't know what editor you use, but VSCode or webstorm should do the right thing out of the box. Feel free to reach out on #docs-infra for help figuring out the editor config.

**Comment by IgorMinar** [comment]
@aikidave last but not least! Awesome job sending a PR and trying to figure out how to change the app!!! I love that you are getting your hands dirty and making changes. That's the way to go. Please do not let the review feedback discourage you!

File: aio/angular.json (hunk 10 above)
Comment (aikidave) [reply aio/angular.json]: I will as soon as I figure out the commands to do so. Ah, the learning curve of git... :)

File: aio/content/navigation.json (hunk 11 above)
Comment (aikidave) [reply aio/content/navigation.json]: I am using VS Code. I'm not sure what's happening there. I'll research!

**Comment by aikidave** [comment]
Thanks for the feedback! I'm not discouraged at all. This is how the process is supposed to work, right?

I do agree that going 4 levels into the navigation is not how I'd like to deal with things; however, I can't think of a better option at the moment.  We want this information discoverable, but if we make it too prominent it gives the impression users have to do a lot of work to update to version 9, and that's not the case!

Regarding the length of the titles in the left nav: I don't really want to reduce the font size. I'll think on this some more and see what I can come up with.

**Comment by aikidave** [comment]
I've pushed my changes to Ajit's PR: #34979. We'll use that one going forward, so please put all comments there. Thanks!
